editor/src/components/inspector/sections/image-section/image-source-control.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { treeToContents } from '../../../../components/assets'
+import { isImageFile } from '../../../../core/model/project-file-utils'
+import { ProjectContents } from '../../../../core/shared/project-file-types'
+import { useEditorState } from '../../../editor/store/store-hook'
+import { useInspectorElementInfo } from '../../common/property-path-hooks'
+import { OptionChainControl } from '../../controls/option-chain-control'
+import { SelectControl, SelectOption } from '../../controls/select-control'
+import { StringControl } from '../../controls/string-control'
+
+enum ImageSrcType {
+  URL = 1,
+  LOCAL,
+}
+
+const getProjectImageFileNames = (projectContents: ProjectContents): string[] => {
+  return Object.keys(projectContents).filter((key) => isImageFile(projectContents[key]))
+}
+
+const isImgSrcLocal = (src: string, localFiles: string[]) => {
+  return src.startsWith('./') && localFiles.includes(src)
+}
+
+const chainControlOptions: Array<SelectOption> = [
+  { value: ImageSrcType.URL, label: 'URL' },
+  { value: ImageSrcType.LOCAL, label: 'Local' },
+]
+
+export const ImageSourceControl = React.memo(() => {
+  const {
+    value: srcValue,
+    controlStyles: srcControlStyles,
+    controlStatus: srcControlStatus,
+    onSubmitValue: srcOnSubmitValue,
+  } = useInspectorElementInfo('src')
+
+  const { projectContents } = useEditorState((store) => {
+    return {
+      projectContents: store.editor.projectContents,
+    }
+  }, 'ImgSection')
+
+  const localImageFilenames = React.useMemo(() => {
+    return getProjectImageFileNames(treeToContents(projectContents)).map((filename) => {
+      // prepending '.' to the absolute path to make it reference project files correctly
+      return `.${filename}`
```

Comment:
I thuink we already have a helper for this https://github.com/concrete-utopia/utopia/blob/master/editor/src/common/server.ts#L69

Reply:
uh noice! thanks for the heads up (https://github.com/concrete-utopia/utopia/pull/2660/commits/c2ebcdece9248608a7977d2cf6d1cb3b78c2499e)